### PR TITLE
[BugFix] Fix `admin show replica distribution` not work correctly with CN deployment in shared-data mode 

### PR DESCRIPTION
**Pull request by wxl24life**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by kangkaisen — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/MetadataViewer.java [hunk 1]
```diff
@@ -235,7 +233,8 @@ private static List<List<String>> getTabletDistribution(String dbName, String tb
             // backend id -> replica count
             Map<Long, Integer> countMap = Maps.newHashMap();
             // init map
-            List<Long> beIds = infoService.getBackendIds(false);
+            SystemInfoService infoService = GlobalStateMgr.getCurrentState().getNodeMgr().getClusterInfo();
+            List<Long> beIds = infoService.getAllComputeNodeIds();
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/catalog/MetadataViewer.java:237]: need take warehouse into consideration in shared mode. cc @HangyuanLiu

File: fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java [hunk 2]
```diff
@@ -867,6 +867,18 @@ public List<Long> getBackendIds(boolean needAlive) {
...
     }
 
+    /**
+     * Common method to get all compute nodes, include be and cn
+     * @return
+     */
+    public List<Long> getAllComputeNodeIds() {
+        List<Long> computeNodeIds = getBackendIds(false);
+        if (RunMode.isSharedDataMode()) {
+            computeNodeIds.addAll(getComputeNodeIds(false));
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java:877]:
```
            long warehouseId = WarehouseManager.DEFAULT_WAREHOUSE_ID;
            if (ConnectContext.get() != null) {
                warehouseId = ConnectContext.get().getCurrentWarehouseId();
            }
            return GlobalStateMgr.getCurrentState().getWarehouseMgr().getAllComputeNodeIds(warehouseId);
```

Comment (wxl24life) [reply fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java]: done

File: fe/fe-core/src/main/java/com/starrocks/catalog/MetadataViewer.java [hunk 3]
```diff
@@ -276,6 +277,23 @@ private static List<List<String>> getTabletDistribution(String dbName, String tb
...
+        if (RunMode.isSharedDataMode()) {
+            // check warehouse
+            long warehouseId = ConnectContext.get().getCurrentWarehouseId();
+            List<Long> computeNodeIs =
+                    GlobalStateMgr.getCurrentState().getWarehouseMgr().getAllComputeNodeIds(warehouseId);
+            if (computeNodeIs.isEmpty()) {
+                Warehouse warehouse = GlobalStateMgr.getCurrentState().getWarehouseMgr().getWarehouse(warehouseId);
+                throw new DdlException("no available compute nodes in warehouse " + warehouse.getName());
+            }
+            allComputeNodeIds.addAll(computeNodeIs);
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/catalog/MetadataViewer.java:292]: will contain duplicate ids, if there are backend nodes there in the system, they will be added in line282 and repeated again in line 292.

**Review by kevincai — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Pull request merged**